### PR TITLE
Adaptive gradient clipping (MSE-to-L1 transition stabilizer)

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
The MSE-to-L1 curriculum transitions at epoch 4. At that point, the loss landscape changes abruptly — MSE gradients are error-proportional while L1 gradients are constant-magnitude. This gradient discontinuity can cause destabilizing parameter updates right when the model needs to refine. Gradient clipping (max_norm) has never been tried in this codebase.

By using an **adaptive clip schedule** — looser during MSE phase (max_norm=2.0) for fast early learning, tighter during L1 phase (max_norm=0.5) for stable refinement — we smooth the transition and prevent any single batch from causing outsized updates.

This is distinct from gradient centralization (PR #276, neutral) which modifies gradient *direction* but not *magnitude*. Clipping constrains magnitude directly.

## Instructions

In `train.py`, between `loss.backward()` and `optimizer.step()` (around lines 152-154), add adaptive gradient clipping:

```python
        optimizer.zero_grad()
        loss.backward()
        # Adaptive gradient clipping: loose during MSE, tight during L1
        alpha = min(epoch / 4.0, 1.0)
        max_norm = 2.0 * (1.0 - alpha) + 0.5 * alpha  # 2.0 -> 0.5
        torch.nn.utils.clip_grad_norm_(model.parameters(), max_norm=max_norm)
        optimizer.step()
```

That's the only change. Everything else stays the same — same loss, same curriculum, same hyperparams.

W&B tag: `mar14b`, group: `grad-clip-adaptive`

## Baseline
- **surf_p = 107.35**, surf_Ux = 1.23, surf_Uy = 0.84, val_loss = 2.45

---

## Results
_To be filled by student_

### CODE REVIEW

**Comment by tcapelle** [comment]
**Closing — results worse than baseline.**

| Metric | Baseline | This PR | Delta |
|--------|----------|---------|-------|
| surf_p | 107.35 | 114.80 | +7.0% ❌ |
| surf_Ux | 1.23 | 1.19 | -3.3% |
| surf_Uy | 0.84 | 0.87 | +3.6% |

Gradient clipping hurt pressure accuracy significantly. The clipping norms (2.0→0.5) may have been too aggressive, constraining useful gradient signal. Moving on to other approaches.
